File: src/main/java/com/qulix/shilomy/trainingtask/web/repository/ProjectRepository.java
package com.qulix.shilomy.trainingtask.web.repository;

import com.qulix.shilomy.trainingtask.web.entity.ProjectEntity;

import java.util.List;

public interface ProjectRepository {
    void addProject(ProjectEntity project);
    void removeProject(ProjectEntity project);
    void updateProject(ProjectEntity project); // Think it as replace for set

    List<?> query(ProjectSpecification specification);
}

File: src/main/kotlin/com/makshilo/trainingtask/repository/ProjectRepository.kt
package com.makshilo.trainingtask.repository

import com.makshilo.trainingtask.model.Project

interface ProjectRepository {
  fun add(project: Project): Project
  fun update(project: Project): Project
  fun remove(id: Long)
  fun findById(id: Long): Project?
  fun findAll(): List<Project>
}
